### PR TITLE
add phpcs ruleset

### PR DESCRIPTION
**Pull request by afragen**
I also removed the ignore at head of plugin.php

### CODE REVIEW

File: phpcs.xml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0"?>
+<ruleset name="Updates API Inspector">
+    <description>Ruleset for Updates API Inspector.</description>
+
+    <!-- Show progress and sniff codes in all reports -->
+    <arg value="ps"/>
+
+    <rule ref="WordPress-Core">
+        <exclude name="WordPress.Files.FileName.InvalidClassFileName" />
```
Comment (pbiron) [inline phpcs.xml]: Actually, `InvalidClassFileName` should not be excluded.  When I get around to doing the refactoring to my normal plugin structure, the main `plugin.php` file will be the only one that violates that...so the inline `phpcs:ignore` in that file is just fine.

File: plugin.php [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<?php // phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
```
Comment (pbiron) [inline plugin.php]: Revert this change, since `WordPress.Files.FileName.InvalidClassFileName` shouldn't be excluded, per the comment above.

**Review by pbiron — CHANGES_REQUESTED** [review]

File: plugin.php (hunk 2 above)
Comment (afragen) [reply plugin.php:1]: Not sure I understand. The inline `phpcs:ignore` is no longer necessary as it's in the ruleset.

**Comment by pedro-mendonca** [comment]
@afragen the ruleset [`WordPress`](https://github.com/WordPress/WordPress-Coding-Standards/blob/develop/WordPress/ruleset.xml) already includes `WordPress-Core`.

```
WordPress
 |_ WordPress-Docs
 |_ WordPress-Extra
      |_ WordPress-Core
```

**Comment by afragen** [comment]
@pbiron need/want anything more for this?

**Comment by pbiron** [comment]
> @pbiron need/want anything more for this?

The one question I have is: what is the difference between setting the `severity` to 0 and `excluding` a sniff?  As in

```
<rule ref="Squiz.PHP.DisallowBooleanStatement.Found">
    <severity>0</severity>
</rule>
```

vs

```
<rule ref="Squiz.PHP">
        <exclude name="Squiz.PHP.NonExecutableCode.ReturnNotRequired"/>
</rule>
```

**Comment by afragen** [comment]
I think the difference is excluding it entirely vs possibly setting the severity to allow under certain circumstances.

I'm happy to change according to your preferences.

**Comment by pbiron** [comment]
> I think the difference is excluding it entirely vs possibly setting the severity to allow under certain circumstances.
> 
> I'm happy to change according to your preferences.

Sorry I wasn't clear: I don't have a preference because I don't understand the pros/cons of doing it the 1st vs the 2nd way.

I think I understand what `<exclude>` does: phpcs will not bother running that named sniff anywhere.

Naively, it seems like setting `<severity>0</severity>` would cause phpcs to actually run the named sniff but not bother to report any violations...which just seems like a waist of CPU, but I don't know.  Which is why I'm asking what the differences are.

**Comment by afragen** [comment]
I found this. https://github.com/WordPress/WordPress-Coding-Standards/pull/1262

Though I don't necessarily agree with it as I have excludes and then reapply the sniff. Though I see the point of decreasing and increasing the severity. I don't think the clock cycles are worth considering.

**Comment by afragen** [comment]
I’ll update to use ‘exclude’ as it seems to make more sense here.

**Comment by afragen** [comment]
@pbiron any more thoughts on this?

**Comment by pbiron** [comment]
> @pbiron any more thoughts on this?

Sorry Andy.  I've been using a slight modification of this phpcs.xml locally for several days and it works great!

It doesn't really make sense to merge this, tho, without also having a full set of "Build Tools" also setup (e.g., composer, npm, phpunit, etc).  I'm close to having all of that setup locally the way I want it, plus a complete rewrite of the plugin code itself to make a better foundation for future versions of the plugin.

It'll probably take me a few more days to get that completed and when I do I'll push it to `develop`.
